mod/thewire/views/default/forms/thewire/add.php

```diff
@@ -16,14 +19,24 @@
 }
 $chars_left = elgg_echo('thewire:charleft');
 
-$parent_input = '';
-if ($post) {
-	$parent_input = elgg_view('input/hidden', [
-		'name' => 'parent_guid',
-		'value' => $post->guid,
+$guid_input = '';
+if ($guid) {
+	$guid_input = elgg_view('input/hidden', [
+		'name' => 'guid',
+		'value' => $guid,
 	]);
 }
 
+$parent_guid = 0;
+
+if ($guid) {
+	$parent_guid = (int) $entity->wire_thread;
```

Comment:
are you sure `get_entity()` succeeded? It would be better to check `if ($entity instanceof ElggWire)`